wolfcrypt/src/aes.c

```diff
@@ -7856,10 +8805,10 @@ int AES_GCM_decrypt_C(Aes* aes, byte* out, const byte* in, word32 sz,
 #ifdef WOLFSSL_PIC32MZ_CRYPT
     if (blocks) {
         /* use intitial IV for PIC32 HW, but don't use it below */
-        XMEMCPY(aes->reg, ctr, AES_BLOCK_SIZE);
+        XMEMCPY(aes->tmp, ctr, AES_BLOCK_SIZE);
```

Comment:
This looks like a merge error and should be reverted along with line 8811 below. I believe this fix comes from PR #1479.

Commit Notes:
This fixes updating the AES-CBC IV for PIC32MZ hardware acceleration when encrypt/decrypt is called multiple times in a row. It also adds a test for the use case.

Reply:
The IV is passed into AES_GCM_decrypt_C() and processed into the initial block counter (initialCounter). That value is copied into the block counter (ctr). (initialCounter is hashed back in later at the end of GHASH.) ctr is the used to generate the key stream. For PIC32MZ, ctr is copied into aes->tmp and that is passed into the Pic32AesCrypt() function.

What I was thinking is for the AesGcmDecrypt, it doesn't matter if the Pic32 code was using reg or tmp. I didn't want to step on reg since I was using that for storing the internally managed IV. I believe the changes to lines 8808 and 8810 are correct.

For AES_GCM_encrypt_C(), I was handwaving the same logic. But I think that's correct there, too.

Reply:
I retract my previous comment. It was in error.